Add default behavior documentation for Set-Cookie Path attribute (#40380)
mdn/content: files/en-us/web/http/reference/headers/set-cookie/index.md

- `Path=<path-value>` {{optional_inline}}
  - : Indicates the path that _must_ exist in the requested URL for the browser to send the `Cookie` header.

>     If omitted, this attribute defaults to the "directory" of the request URI's path component. As per [RFC 6265](https://datatracker.ietf.org/doc/html/rfc6265#section-4.1.2.4), if the server omits the Path attribute, the user agent will use the "directory" of the request-uri's path component as the default value. For example, if a cookie is set by a request to `/docs/Web/HTTP/index.html`, the default path would be `/docs/Web/HTTP/`.

Reviewer: We don't specifically quote the spec as justification in text. Also, the example is better with an absolute URL since that shows the host is omitted.

```suggestion
    If omitted, this attribute defaults to the path component of the request URL. For example, if a cookie is set by a request to `https://example.com/docs/Web/HTTP/index.html`, the default path would be `/docs/Web/HTTP/`.
```

>     The user agent will include the cookie in an HTTP request only if the path portion of the request-uri matches (or is a subdirectory of) the cookie's Path attribute, where the forward slash (`/`) character is interpreted as a directory separator.
> 

Reviewer: Reverting this spec language. Doesn't say anything more than the text up the top of the section "Indicates the path that _must_ exist in the requested URL for the browser to send the `Cookie` header."

```suggestion
```

    The forward slash (`/`) character is interpreted as a directory separator, and subdirectories are matched as well. For example, for `Path=/docs`,
    - the request paths `/docs`, `/docs/`, `/docs/Web/`, and `/docs/Web/HTTP` will all match.
    - the request paths `/`, `/docsets`, `/fr/docs` will not match.

    > [!NOTE]
    > The `path` attribute lets you control what cookies the browser sends based on the different parts of a site.

>     > It is not intended as a security measure, and [does not protect](/en-US/docs/Web/API/Document/cookie#security) against unauthorized reading of the cookie from a different path. Although seemingly useful for isolating cookies between different paths within a given host, the Path attribute cannot be relied upon for security (see [RFC 6265 Section 8](https://www.rfc-editor.org/rfc/rfc6265#section-8)).

Reviewer: Reverting this for same reason above - we're not duplicating the spec or linking to the spec. We're rewriting the spec in human readable form. The original does this better

```suggestion
    > It is not intended as a security measure, and [does not protect](/en-US/docs/Web/API/Document/cookie#security) against unauthorized reading of the cookie from a different path.
```
